src/google/protobuf/descriptor.cc

```diff
@@ -2612,10 +2612,12 @@ void FileDescriptor::CopyHeadingTo(FileDescriptorProto* proto) const {
   }
 
   // TODO(liujisi): Also populate when syntax="proto2".
```

Comment:
I think you're missing https://github.com/protocolbuffers/protobuf/commit/9b218c1d9aaa8108f65cca0d65325936a8bbeac6